### PR TITLE
Implement new architecture for tlogserver, vm, nbd with backup storage

### PR DESCRIPTION
**Pull request by khaledkbadr**
relates to #471

### CODE REVIEW

File: templates/tlogserver/actions.py [hunk 1]
```diff
@@ -66,24 +82,30 @@ def install(job):
...
+                             log=logpath)
+            if backup:
+                cmd += '-with-slave-sync'
+            container.client.system(cmd)
             if not is_port_listening(container, port):
                 raise j.exceptions.RuntimeError('Failed to start tlogserver {}'.format(service.name))
             service.model.data.bind = '%s:%s' % (ip, port)
             container.node.client.nft.open_port(port)
+        else:
+            # send a siganl sigub(1) to reload the config in case it was changed.
```
Comment (grimpy) [inline templates/tlogserver/actions.py]: fix comment and use int(signal.SIGHUB)

File: templates/vm/actions.py [hunk 2]
```diff
@@ -89,12 +91,23 @@ def _nbd_url(container, nbdserver, vdisk):
...
+    import random
     service = job.service
 
     # creates all nbd servers for each vdisk this vm uses
     job.logger.info("creates vdisks container for vm {}".format(service.name))
-    vdisk_container = create_zerodisk_container(job, service.parent)
-    _init_zerodisk_services(job, vdisk_container)
+    services = service.aysrepo.servicesFind(role="node")
+
+    node = random.choice(services)
```
Comment (grimpy) [inline templates/vm/actions.py]: if there are miltiple nodes pop yourself and do random

File: templates/vm/actions.py [hunk 3]
```diff
@@ -218,6 +231,21 @@ def stop(job):
     cleanupzerodisk(job)
 
 
+def destroy(job):
+    stop(job)
+    service = job.service
+    tlogservers = service.producers.get('tlogserver', None)
```
Comment (grimpy) [inline templates/vm/actions.py]: default to empty list not None otherwise loop will crash

File: templates/vm/actions.py (hunk 2 above)
Comment (khaledkbadr) [inline templates/vm/actions.py]: That's what I do :p

**Pull request merged**